src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstafflookupservice/service/IndexingService.kt

```diff
@@ -5,12 +5,16 @@ import uk.gov.justice.digital.hmpps.hmppsstafflookupservice.client.MicrosoftGrap
 
 @Service
 class IndexingService(
-  val microsoftGraphClient: MicrosoftGraphClient
+  val microsoftGraphClient: MicrosoftGraphClient,
+  val databaseWriteService: DatabaseWriteService
 ) {
   suspend fun indexAll(): String {
-    val exampleData = microsoftGraphClient.getUsersWithoutPagination()
-    exampleData.collect {
-      println("DATA: $it")
+    var graphResponse = microsoftGraphClient.getUsersPage(null)
+    databaseWriteService.writeData(graphResponse.value)
+    while (graphResponse.nextLink != null) {
+      val skipToken = graphResponse.nextLink!!.substringAfter("skiptoken=")
```

Comment:
is there ever a time when there isn't a skiptoken in the nextLink? Is it always at the end?

Reply:
The response for the last page does not contain a nextLink (I tested that). If there isn't a nextLink then it won't get here..